src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java

```diff
@@ -124,7 +125,11 @@ public boolean verifyCertificate(X509Certificate certificate, Attributes attribu
             Attribute attribute = attributes.get(ldapAttribute);
             final int size = attribute.size();
             for (int i = 0; i < size; i++) {
-                if (certificate.getSubjectDN().getName().equals(attribute.get(i))) {
+
+                X500Principal certSubjectX500Principal = certificate.getSubjectX500Principal();
+                X500Principal attSubjectX500Principal = new X500Principal((String)attribute.get(i));
```

Comment:
@darranl I have updated the PR including a null check for the different attribute values used in this class. The user is able to configure any name for mapping the user certificate attributes, so he could configure the values to a LDAP attribute that, being present in the LDAP server, could return a null value.

About casting type checks, the LDAP context provider used only works with two types of values `String` and `byte[]`.`EncodedCertificateVerifier` is the only one managing binary data, because it is the only one overriding the `addBinaryLdapAttributes` method, so IMO it is unnecessary to check the type before convert.